src/header/js/header-menus.js

```diff
@@ -220,21 +229,63 @@ export default ({variant = 'desktop'}) => {
 	});
 
 	function renderSingle({hasMenuItems, $subcontent, item, uuid, suuid}) {
-		if (hasMenuItems) {
-			$($subcontent,
-				$('ul',
-					{
-						class: `${prefix}-sublist`,
-						role: 'navigation', aria: {labelledby: `${prefix}-link-${variant}-${uuid}-${suuid}`}
-					},
-					/* Global Navigation: Menus: Sublink
-					/* ============================== */
-					$('div', {class: `${prefix}-sublist--col-wrapper`},
-						createMenuColumns(item.menus.slice(0, 9)),
-						createMenuColumns(item.menus.slice(9, 18))
+		if (item.menus.length >= 10) {
+			if (item.menus.length % 2 === 0) {
+				const multi = item.menus.length / 2;
+				
+				if (hasMenuItems) {
+					$($subcontent,
+						$('ul',
+							{
+								class: `${prefix}-sublist`,
+								role: 'navigation', aria: {labelledby: `${prefix}-link-${variant}-${uuid}-${suuid}`}
+							},
+							/* Global Navigation: Menus: Sublink
+							/* ============================== */
+							$('div', {class: `${prefix}-sublist--col-wrapper`},
+								createMenuColumns(item.menus.slice(0, multi)),
+								createMenuColumns(item.menus.slice(parseInt(multi), item.menus.length - 1))
```

Comment:
This is repetitive. The only lines that change, within these if statements, are the calls to createMenuColumns. The $subcontent, ul, and div elements are the same. Could you think of a way to refactor this?

```js

getMenuColumns(menus){


  if (menus.length % 3 === 0) {
  	const multi = item.menus.length / 2;
	return $( 
	  createMenuColumns(item.menus.slice(0, multi)),
	  createMenuColumns(item.menus.slice(parseInt(multi), parseInt(multi * 2))),
	  createMenuColumns(item.menus.slice(parseInt(multi * 2), item.menus.length - 1))
	);
  }

// ... etc
```

Reply:
@gschnall this has been refactored..